sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/JoinOptimizationSuite.scala

```diff
@@ -145,4 +159,15 @@ class JoinOptimizationSuite extends PlanTest {
     }
     assert(broadcastChildren.size == 1)
   }
+
+  test("SPARK-23172 skip projections when flattening joins") {
```

Comment:
Could you add a test case which would fail before the fix?

Reply:
ok